cmd/eksctl-anywhere/cmd/generatehardware.go

```diff
@@ -62,105 +54,90 @@ func init() {
 	}
 }
 
-func preRunGenerateHardware(cmd *cobra.Command, args []string) {
-	cmd.Flags().VisitAll(func(flag *pflag.Flag) {
-		err := viper.BindPFlag(flag.Name, flag)
-		if err != nil {
-			log.Fatalf("Error initializing flags: %v", err)
-		}
-	})
-}
-
 func (hOpts *hardwareOptions) generateHardware(ctx context.Context) error {
-	if !hOpts.dryRun && hOpts.tinkerbellIp == "" {
-		return errors.New("tinkerbell-ip is required, please specify it using --tinkerbell-ip")
+	if err := validateOptions(hOpts); err != nil {
+		return err
 	}
 
-	csv, err := hardware.NewCsvParser(hOpts.csvPath)
+	csvFile, err := os.Open(hOpts.csvPath)
 	if err != nil {
-		return err
+		return fmt.Errorf("csv: %v", err)
 	}
 
-	defer csv.Close()
+	outputDir, err := hardware.CreateDefaultDir()
```

Comment:
Did you get rid of the outputPath cli option to override output of the harware-manifest directory to a different location besides always going into working dir?

Reply:
Please pass it to CreateDefaultDir so we can still support this functionality.

Reply:
Wowzers, I hadn't spotted that, thanks.